src/Course/State.hs

```diff
@@ -247,7 +268,7 @@ memberAndAddSetN x set = (S.member x set, S.insert x set)
 distinct :: Ord a => List a -> List a
 distinct xs = eval (filtering p xs) S.empty
   where
-    p x = State $ \set -> (S.member x set, S.insert x set)
+    p x = State $ \set -> (not $ S.member x set, S.insert x set)
```

Comment:
Interesting...

Reply:
whaaa? how was this working before? lol :)